src/pages/studyView/SummaryHeader.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Sample } from 'shared/api/generated/CBioPortalAPIInternal';
+import { observer } from "mobx-react";
+import { computed, observable, action } from 'mobx';
+import "./styles.scss";
+import { bind } from 'bind-decorator';
+import { buildCBioPortalUrl } from 'shared/api/urls';
+import CustomCaseSelection from 'pages/studyView/customCaseSelection/CustomCaseSelection';
+import { SingleGeneQuery } from 'shared/lib/oql/oql-parser';
+
+export interface ISummaryHeaderProps {
+    selectedSamples: Sample[];
+    updateCustomCasesFilter:(samples:Sample[]) => void;
+}
+
+@observer
+export default class SummaryHeader extends React.Component<ISummaryHeaderProps, {}> {
+
+    @observable private isCustomCaseBoxOpen = false;
+
+    @computed
+    get selectedPatientsCount() {
+        return _.uniq(this.props.selectedSamples.map(sample => sample.uniquePatientKey)).length;
+    }
+
+    @bind
+    private openCases() {
+
+        const firstSample = this.props.selectedSamples[0];
+        const groupedSamples = _.groupBy(this.props.selectedSamples, sample => sample.studyId);
+        const IncludeStudyId: boolean = Object.keys(groupedSamples).length > 1;
+
+        let navCaseIds = _.map(groupedSamples, (samples) => {
+            return _.map(samples, sample => {
```

Comment:
I dont see the point of computing groupedSamples?

Reply:
Here its used to check whether `studyId` needs to be prefixed or not. and also to order samples by study

Reply:
ah yes, makes sense